pontoon/base/models/user.py

```diff
@@ -234,19 +235,62 @@ def badges_translation_count(self):
 @property
 def badges_review_count(self):
     """Translation reviews provided by user that count towards their badges."""
-    return self.actions.filter(
-        Q(action_type="translation:approved") | Q(action_type="translation:rejected"),
+    approved_reviews = self.actions.filter(
+        action_type="translation:approved",
         created_at__gte=settings.BADGES_START_DATE,
-    ).count()
+    )
+
+    rejected_reviews = self.actions.filter(
+        action_type="translation:rejected",
+        created_at__gte=settings.BADGES_START_DATE,
+    )
+
+    # Exclude auto-rejections caused by creating a new translation or approving an existing one
+    rejected_reviews = rejected_reviews.exclude(
+        Exists(
+            self.actions.filter(
+                performed_by=OuterRef("performed_by"),
+                action_type__in=["translation:created", "translation:approved"],
+                created_at__gt=OuterRef("created_at"),
+                created_at__lte=OuterRef("created_at") + timedelta(milliseconds=100),
+            )
+        )
+    )
+
+    return approved_reviews.count() + rejected_reviews.count()
```

Comment:
You could reduce the number of DB queries like this:

```python
    closely_preceded_created = ActionLog.objects.filter(
        performed_by=OuterRef("performed_by"),
        ...
    )

    return self.actions.filter(
        Q(action_type="translation:approved") |
        Q(
            action_type="translation:rejected",
            ~Exists(closely_preceded_direct_action)
        ),
        created_at__gte=settings.BADGES_START_DATE,
    ).count()
```

Reply:
would switching the order of the `~Exists` like this cause poor performance? If I keep as the second argument then pylance throws an error that "Positional argument cannot appear after keyword arguments"

```
return self.actions.filter(
        Q(action_type="translation:approved") |
        Q(  
            ~Exists(closely_preceded_action),
            action_type="translation:rejected",
        ),
        created_at__gte=settings.BADGES_START_DATE,
    ).count()
```

Reply:
No, that shouldn't have performance implications.